### ISSUE TITLE
fixing [Error during WebSocket handshake: Unexpected response code: 403] when serving html from different server

### ISSUE THREAD

**Issue by sanskarsharma** [gorilla/websocket#644]
**Describe the problem you're having**
I have created a chat server following https://github.com/gorilla/websocket/blob/master/examples/chat with some modifications. I want to serve the html file (or any other static asset) from a different front-end server.
It is working fine when I am serving `index.html` from the same go server, but when using a different server to serve index.html, I am getting the this error : 
![image](https://user-images.githubusercontent.com/20902129/96371996-5dad6f00-1182-11eb-87b3-ee887e88c45e.png)

There are no logs on the go server when this happens.

few things I have tried : 
- I have tried setting `CheckOrigin` in the upgrade assignment
```go
var upgrader = websocket.Upgrader{
	ReadBufferSize:    4096,
	WriteBufferSize:   4096,
	EnableCompression: true,
	CheckOrigin: func(r *http.Request) bool {
		return true
	},
}
```

- I am using gin, so have tried allowing all origins using https://github.com/gin-contrib/cors

**Versions**
> Go version: go1.14.2 darwin/amd64
> package version: github.com/gorilla/websocket v1.4.2
…

**"Show me the code!"**
javascript websocket initialization
```javascript
const params = window.location.href.split("/");

const roomId = params[params.length - 1];
conn = new WebSocket("ws://0.0.0.0:8080/ws/" + getParameterByName("room_id"));
```
here `getParameterByName` just returns value of key from query params

go ws api
```go
router.GET("/ws/:roomId", func(c *gin.Context) {
		roomId := c.Param("roomId")
		log.Println(roomId)
		serveWs(c.Writer, c.Request, roomId)
})
```

```go
func serveWs(w http.ResponseWriter, r *http.Request, roomId string) {
	ws, err := upgrader.Upgrade(w, r, nil)
	if err != nil {
		log.Println(err.Error())
		return
	}
	subscriber := &Subscriber{send: make(chan Message, 256), connection: ws}
	roomSubscription := RoomSubscription{subscriber, roomId}
	h.register <- roomSubscription
	go roomSubscription.writePump()
	go roomSubscription.readPump()
}
```
…

I have gone through a similar issue reported but the fix there didn't help my case : https://github.com/gorilla/websocket/issues/233

Any help would be hugely appreciated, Thanks !

**Comment by TaylorRex** [comment]
The client is not connecting to websocket endpoint shown in the issue.  

- The websocket package does not return 403 when the check origin function is `func(r *http.Request) bool { return true }`. Something else is returning this response.
- The websocket endpoint logs the room id and errors, yet no logs are visible.

The problem is with the client code or the HTTP server configuration.

**Comment by sanskarsharma** [comment]
@TaylorRex As you pointed out, the issue was with the HTTP server config, hidden by my own stupidity, explaining below.

I had left the same app running on same port in a docker container on local and forgot to stop that while I was testing by making changes and running `go run`. So the changes were not reflecting while I was testing and I was constantly seeing the app being served by docker. Strangely I didn't get any `port in use` error during this.
Spent ~6 hours scratching my head before I saw the app running without `go run` 😅 

What fixed it :Using https://github.com/gin-contrib/cors with default config

Thanks for the quick response and the help @TaylorRex ! 
Closing this
